cmd/pull.go

```diff
@@ -15,36 +15,45 @@
 package cmd
 
 import (
-	"fmt"
+	"strings"
 
 	"github.com/spf13/cobra"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/viper"
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/client"
+	"golang.org/x/net/context"
 )
 
-// pullCmd represents the pull command
 var pullCmd = &cobra.Command{
 	Use:   "pull",
-	Short: "A brief description of your command",
-	Long: `A longer description that spans multiple lines and likely contains examples
-and usage of using your command. For example:
-
-Cobra is a CLI library for Go that empowers applications.
-This application is a tool to generate the needed files
-to quickly create a Cobra application.`,
+	Short: "Pull KDK docker image",
+	Long:  `Pull the latest/configured KDK docker image`,
 	Run: func(cmd *cobra.Command, args []string) {
-		fmt.Println("pull called")
+		logger := logrus.New().WithField("command", "pull")
+
+		imageCoordinates := []string{viper.Get("image.repository").(string), viper.Get("image.tag").(string)}
+
+		ctx := context.Background()
+		cli, err := client.NewEnvClient()
+
+		if err != nil {
+			logger.WithField("error", err).Fatal("Failed to create docker client")
+		}
+
+		logger.Info("Pulling KDK image. This may take a few minutes...")
+
+		out, err := cli.ImagePull(ctx, strings.Join(imageCoordinates, ":"), types.ImagePullOptions{})
+		if err != nil {
+			logger.WithField("error", err).Fatal("Failed to pull KDK image")
+			panic(err)
+		}
+		defer out.Close()
```

Comment:
I think this defer needs to move up a few lines before L47: if err != nil.

Reply:
good catch!